### DISCUSSION TITLE
Multipart Body configuration on actions

### DISCUSSION THREAD

**Discussion by AashishDhakal** [drakkan/sftpgo#1494] [Q&A] [answered]
Can anyone point me out to the multipart configuration docs or may be share the correct way to set it up if tried it before? 

I basically want to attach file to the HTTP request.

**Comment by nicholas-coker** [comment]
Were you able to find a solution? We are trying to do the same thing and the documentation does not help. Thanks!

**Comment by AashishDhakal** [comment, marked as answer]
@nicholas-coker We did a FastAPI middleware, here's how we have implemented.

```
from pathlib import Path
from fastapi import Body

@app.post("/upload/")
def upload(upload_body=Body()):
    username = upload_body.get("username")
    file_path = upload_body.get("path")
    path_obj = Path(file_path)
    ....
```

**Comment by nicholas-coker** [comment]
Awesome, thanks! We figured out a way to do it running a bash script:

ACTION
Name:			CURL-API
Type:			Command
Command:		/etc/sftpgo/curl_script.sh
Arguments:		{{Name}},{{FsPath}}
Timeout:		20

RULE
Trigger:			Filesystem events
Fs events:		upload
Actions:			CURL-API

OS
Edit:			/etc/sftpgo/curl_script.sh
```
#!/bin/bash
/usr/bin/curl --header “key: data” -F “username=$1” -F “file=@$2” API-URL
```
